### PR TITLE
Call API to import a Google Classroom on "Import Section" click

### PR DESCRIPTION
**Pull request by joshlory**
Connects the "Import Section" button to `ApiController#import_google_classroom`.  Follow up to PR https://github.com/code-dot-org/code-dot-org/pull/16342.

screen shot 2017-07-12 at 2 16 56 pm

### CODE REVIEW

File: apps/src/templates/teacherDashboard/shapes.jsx [hunk 1]
```diff
@@ -1,11 +1,12 @@
 import { PropTypes } from 'react';
+import { SectionLoginType } from '@cdo/apps/util/sharedConstants';
 
 export const sectionShape = PropTypes.shape({
   id: PropTypes.number.isRequired,
   name: PropTypes.string.isRequired,
   // Though we validate valid login types here, the server actually owns the
   // canonical list, and passes us the list of valid login types.
-  loginType: PropTypes.oneOf(['word', 'email', 'picture']).isRequired,
+  loginType: PropTypes.oneOf(Object.keys(SectionLoginType)).isRequired,
```
Comment (Bjvanminnen) [inline apps/src/templates/teacherDashboard/shapes.jsx:9]: 👍

File: apps/src/templates/teacherDashboard/SectionsPage.jsx [hunk 2]
```diff
@@ -71,10 +72,22 @@ class SectionsPage extends Component {
...
-    // TODO (josh): use `selectedId`.
-    console.log(selectedId);
+  };
+
+  handleImport = courseId => {
+    this.props.importClassroomStarted();
+
+    $.getJSON('/dashboardapi/import_google_classroom', { courseId }).then(() => {
+      this.setState({rosterDialogOpen: false});
+      this.setState({sectionsLoaded: false});
```
Comment (Bjvanminnen) [inline apps/src/templates/teacherDashboard/SectionsPage.jsx]: These two can be combined into one line (i.e. you can set multiple state field in a single call to setState)

File: apps/src/templates/teacherDashboard/SectionsPage.jsx [hunk 3]
```diff
@@ -71,10 +72,22 @@ class SectionsPage extends Component {
...
+
+      $.getJSON("/v2/sections/").done(results => {
+        this.props.setSections(results, true);
+        this.setState({sectionsLoaded: true});
+      });
```
Comment (Bjvanminnen) [inline apps/src/templates/teacherDashboard/SectionsPage.jsx:88]: Both of these APIs set sectionsLoaded. If the call to /v2/sections finishes first, we end up with `sectionsLoaded = true`. If `import_google_classroom` finishes first, we end up with `sectionsLoaded = false`

File: apps/src/templates/teacherDashboard/teacherSectionsRedux.js [hunk 4]
```diff
@@ -18,7 +18,7 @@ export const setValidAssignments = (validCourses, validScripts) => ({
   validCourses,
   validScripts
 });
-export const setSections = sections => ({ type: SET_SECTIONS, sections });
+export const setSections = (sections, reset = false) => ({ type: SET_SECTIONS, sections, reset });
```
Comment (Bjvanminnen) [inline apps/src/templates/teacherDashboard/teacherSectionsRedux.js]: Would like to see a comment somewhere about what `reset` means

**Comment by Bjvanminnen** [comment]
Would like to see some tests.

File: apps/src/templates/teacherDashboard/SectionsPage.jsx (hunk 3 above)
Comment (joshlory) [reply apps/src/templates/teacherDashboard/SectionsPage.jsx:88]: Not sure I follow.  The JSON calls are nested, so the `/v2/sections` request won't start until after the first request resolves.

Comment (Bjvanminnen) [reply apps/src/templates/teacherDashboard/SectionsPage.jsx:88]: My bad. Didn't realize they were nested.

File: apps/src/templates/teacherDashboard/teacherSectionsRedux.js [hunk 5]
```diff
@@ -114,9 +121,10 @@ export default function teacherSections(state=initialState, action) {
   if (action.type === SET_SECTIONS) {
     const sections = action.sections.map(section =>
       sectionFromServerSection(section));
+    const prevSectionIds = action.reset ? [] : state.sectionIds;
     return {
       ...state,
-      sectionIds: state.sectionIds.concat(sections.map(section => section.id)),
+      sectionIds: prevSectionIds.concat(sections.map(section => section.id)),
```
Comment (Bjvanminnen) [inline apps/src/templates/teacherDashboard/teacherSectionsRedux.js:127]: might be good to get some test coverage of this reset behavior in teacherSectionsReduxTest.js

**Comment by Bjvanminnen** [comment]
Was hoping to still see some test coverage of reset behavior in teacherSectionsRedux.

**Comment by joshlory** [comment]
Ok, I'll create a follow up PR.

**Pull request merged**